proposals/lip-0048.md

```diff
@@ -25,15 +25,65 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
+
+This method can be invoked by other modules to charge an additional fee when a state increase is triggered during the execution of a transaction, e.g., when a new [validator is registered][lip-0057#execution] in the PoS module, a new [sidechain is registered][lip-0043#execution], or a [user account is initialized][lip-0051#initializeuseraccountinternal] in the Token module: 
+```python
+Fee.payFee(USER_ACCOUNT_INITIALIZATION_FEE)
+```
+In addition, developers may decide to call the `payFee` method for transactions that require extensive computation, e.g., when verifying an inclusion proof, or if any other resource is extensively used during transaction execution.
+
+#### Transaction Fee Processing
+
+We refer to the sum of the minimum fee and the additional fees processed by the `payFee` method as the "consumed" part of the fee. The remainder of the fee, called the "available" part of the fee, is transferred to the validator as a reward. In this way, validators are assumed to prioritize transactions with a higher available fee in case of network congestion. [After the transaction is executed](#after-command-execution), consumed fees are either burned or transferred to a dedicated pool address. 
+
+The option to transfer the consumed fees to a dedicated pool address `ADDRESS_FEE_POOL` could be beneficial for sidechains, as fees collected this way can be used as an incentive for validators and other protocol participants for various purposes: Funding a community pool for on-chain governance spending, sharing funds proportionally among validators per round, incentivizing ecosystem developers, supporting chain maintenance and improvement, rewarding users' staking and voting activities, etc.
+
+In the Lisk mainchain, the consumed fees are always burned. This reduces the total supply, creating a deflationary pressure that works against inflation.
+
+### Cross-chain Message Fees
+
+Cross-chain messages (CCMs) facilitate interoperability by enabling cross-chain transfer of data. Each CCM contains a [*message fee*][lip-0049#ccmschema], intended to cover the cost of transaction processing on the receiving chain (see [LIP 0049][lip-0049#fee]). 
+
+Since managing message fees requires checking and updating the escrow balances for the message fee token, this task is assigned to the Token module whose [`payMessageFee`][lip-0051#paymessagefee] method charges message fees in the sending chain. In the receiving chain, the Token module assigns the message fee to the relayer account and updates escrows if necessary, and then the Fee module handles the transaction fee processing as described above. Note that the CCM sender should include any additional fees necessary for the CCM execution, e.g., to pay for the user account initialization on the receiving chain.
```

Comment:
```suggestion
Since managing message fees requires checking and updating the escrow balances for the message fee token, this task is assigned to the Token module whose [`payMessageFee`][lip-0051#paymessagefee] method charges message fees in the sending chain. In the receiving chain, the Token module assigns the message fee to the relayer account and updates escrows if necessary, and then the Fee module handles the message fee processing in a similar way as it handles transaction fees.
```

Reply:
Oh. Thanks for noticing this